GTBitsOfGood/southface
# ah/97-add-standard

## ah/97-add-standard

Issue Number(s): #97

- new admin-protected `add-standard` page
- view + edit versions of form, which uses React Final Form to manage state
- standard component library based on React Final Form
- boilerplate for displaying added tags (waiting on designs)

### Checklist

- [x] Created a new page called `add-standard`
- [x] ability to add Title, Criteria, Upload Images (see other notes), select BuildingType, and select primary category
- [x] onClick of "Review and Add" button, it shows the completed fields
- [x] if "Review and Add" button is clicked without entering all relevant fields, show error text (don't redirect basically)
- [x] onClick of "Add to Digital Library" it shows modal showing "are you sure you want to add..."
- [x] onClick of "Yes, add standard" it redirects back to the `add-standard` page which now has a little toast showing "[standard name] added to library"  -- didn't do this exactly but followed the figma design to make a popup element
- [x] all relevant database queries/mutations are made: image, criteria, tags, title are updated on the database.


### How to Test

- Admin view: can see "Add a New Standard" navlink, can go to `/add-standard`
- Non-admin: no navlink, error message on `/add-standard`
- Must fill out all fields except for tag to pass Review & Add
- Cards properly added to database
- Image upload/delete works for any file and shows static image
- Can click "View" after adding a card to go to relevant library route
